tsdb/ooo_head.go

```diff
@@ -36,6 +36,18 @@ func NewOOOChunk() *OOOChunk {
 // Insert inserts the sample such that order is maintained.
 // Returns false if insert was not possible due to the same timestamp already existing.
 func (o *OOOChunk) Insert(t int64, v float64) bool {
+	// Although out-of-order samples can be out-of-order amongst themselves, we
+	// are opinionated and expect them to be usually in-order meaning we could
```

Comment:
Suggest the narrative can go in commit message, and stick to the facts in comment.

Reply:
I think it's fine in the comment so someone else does not revert it

Reply:
I personally like this comment as well. When I first went through this method I just solved it in the natural way of "finding the place where i should put it and then placing it" but today we're opinionated about how we can optimize this based on usage and although literature will be in the commit (already is) I'd rather leave a small paragraph for context.